src/main/java/com/redhat/devtools/lsp4ij/LanguageServiceAccessor.java

```diff
@@ -149,6 +150,11 @@ public CompletableFuture<List<LanguageServerItem>> getLanguageServers(@NotNull V
             return CompletableFuture.completedFuture(Collections.emptyList());
         }
 
+        // getLanguageServers is generally called in Read Access
+        // we get here Document instance to avoid creating a new Read Action thread to get the document from the file.
+        // The document instance is only used when file is opened to add some listener.

```

Comment:
// getLanguageServers is generally called with Read Access.
// We get the Document instance now, to avoid creating a new Read Action thread to get it from the file.
// The document instance is only used when the file is opened, to add some listeners.

Reply:
fixed